libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/MapboxDirectionsSession.kt

```diff
@@ -57,6 +61,49 @@ class MapboxDirectionsSession(
         })
     }
 
+    override fun buildAdjustedRouteOptions(
```

Comment:
I copy pasted this function over here. This has enough logic that it could use its own focused tests and visibility